packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -318,25 +318,16 @@ class FlutterPlugin implements Plugin<Project> {
             pluginProject.android.buildTypes {
                 "${buildType.name}" {}
             }
-            // The embedding is a compileOnly dependency of a Flutter plugin,
-            // however prior to Gradle 6.0.0, it must be an API dependency.
+            // The embedding is API dependency of the plugin, so the AGP is able to desugar
+            // default method implementations when the interface is implemented by a plugin.
             //
-            // Not doing so, causes transitive dependency resolution conflicts.
-            // That is, the embedding dependencies resolved in the plugin are
-            // different than the ones resolved in the app.
-            if (isGradleVersionGraterOrEqualThan('6.0.0')) {
```

Comment:
nit: just noticed this but that method should be `Greater`, not `Grater`.

Reply:
ahh. I removed this method as it's no longer used.